SOURCE: Referred from higher version of mysql-connector-c++ to address the below issues
error: ISO C++17 does not allow dynamic exception specifications
   39 |         void* operator new(size_t size) throw (std::bad_alloc) { return ::operator new(size); }

--- mysql-connector-c++-1.1.9/cppconn/exception.h	2025-03-31 09:12:00.520631259 +0100
+++ mysql-connector-c++-1.1.9/cppconn/exception.h	2025-03-31 09:10:30.881314701 +0100
@@ -35,6 +35,7 @@
 namespace sql
 {
 
+#if (__cplusplus < 201103L)	
 #define MEMORY_ALLOC_OPERATORS(Class) \
 	void* operator new(size_t size) throw (std::bad_alloc) { return ::operator new(size); }  \
 	void* operator new(size_t, void*) throw(); \
@@ -43,7 +44,17 @@
 	void* operator new[](size_t, void*) throw(); \
 	void* operator new[](size_t, const std::nothrow_t&) throw(); \
 	void* operator new(size_t N, std::allocator<Class>&);
+#else
+#define MEMORY_ALLOC_OPERATORS(Class) \
+  void* operator new(size_t size){ return ::operator new(size); }  \
+  void* operator new(size_t, void*) noexcept; \
+  void* operator new(size_t, const std::nothrow_t&) noexcept; \
+  void* operator new[](size_t); \
+  void* operator new[](size_t, void*) noexcept; \
+  void* operator new[](size_t, const std::nothrow_t&) noexcept; \
+  void* operator new(size_t N, std::allocator<Class>&);
 
+#endif
 #ifdef _WIN32
 #pragma warning (disable : 4290)
 //warning C4290: C++ exception specification ignored except to indicate a function is not __declspec(nothrow)
